sdk/rust/src/key_value.rs

```diff
@@ -61,6 +64,31 @@ impl Store {
     pub fn get_keys(&self) -> Result<Vec<String>, Error> {
         key_value::get_keys(self.0)
     }
+
+    #[cfg(feature = "serde")]
+    /// Serialize the given data to JSON, then set it as the value for the specified key.
+    pub fn set_json<T>(&self, key: impl AsRef<str>, value: &T) -> Result<(), anyhow::Error>
+    where
+        T: ?Sized + Serialize,
+    {
+        Ok(key_value::set(
+            self.0,
+            key.as_ref(),
+            &serde_json::to_vec(value)?,
+        )?)
+    }
+
+    #[cfg(feature = "serde")]
+    /// Deserialize an instance of type `T` from the value of `key`.
+    pub fn get_json<T>(&self, key: impl AsRef<str>) -> Result<T, anyhow::Error>
+    where
+        T: ?Sized + for<'de> Deserialize<'de>,
```

Comment:
Do we want these bounds? I would have expected this to be `T: DeserializedOwned` as it doesn't really make sense to return something that has a lifetime tied to the result from `key_value::get` as we drop that at the end of this method. I'm also struggling to think of a use cased to have an unsized type returned from this method.

Reply:
Updated to `T: DeserializeOwned`.